src/nccl_ofi_rdma.c

```diff
@@ -4245,6 +4592,71 @@ static int insert_rdma_send_req_into_msgbuff(nccl_net_ofi_rdma_send_comm_t *s_co
 	return 0;
 }
 
+static int post_rma_write_op(nccl_net_ofi_rdma_req_t *req)
+{
+	nccl_net_ofi_rdma_send_comm_t *s_comm = (nccl_net_ofi_rdma_send_comm_t *)req->comm;
+	nccl_net_ofi_rdma_send_comm_rail_t *comm_rail = get_send_comm_rail(s_comm, 0);
+	rdma_req_rma_op_data_t *rma_op_data = get_rma_op_data(req, NCCL_OFI_RDMA_WRITE);
+	struct fid_mr *rail_mr_handle = rma_op_data->buff_mr_handle->mr[0];
+	void *desc = fi_mr_desc(rail_mr_handle);
+
+	ssize_t rc;
+	/* Post RMA write */
+	rc = fi_write(comm_rail->local_ep, rma_op_data->buff,
+		      rma_op_data->buff_len, desc,
+		      comm_rail->remote_addr,
+		      rma_op_data->remote_buff,
+		      rma_op_data->remote_mr_key, req);
+
+	if ((rc != 0) && (rc != -FI_EAGAIN)) {
+		NCCL_OFI_WARN("fi_write failed; RC: %zd, Error: %s",
+			      rc, fi_strerror(-rc));
+	}
+
+	return rc;
+}
+
+static int post_rma_write_data_op(nccl_net_ofi_rdma_req_t *req)
+{
+	nccl_net_ofi_rdma_send_comm_t *s_comm = (nccl_net_ofi_rdma_send_comm_t *)req->comm;
+	nccl_net_ofi_rdma_send_comm_rail_t *comm_rail = get_send_comm_rail(s_comm, 0);
+	rdma_req_rma_op_data_t *rma_op_data = get_rma_op_data(req, NCCL_OFI_RDMA_WRITE_DATA);
+	ssize_t rc;
+
+	struct iovec iov;
+	struct fi_msg_rma msg;
+	struct fi_rma_iov rma_iov;
+
+	/* Set up the iovec */
+	iov.iov_base = rma_op_data->buff;
+	iov.iov_len = rma_op_data->buff_len;
+
+	/* Set up the rma_iov */
+	rma_iov.addr = rma_op_data->remote_buff;
+	rma_iov.len = rma_op_data->buff_len;
+	rma_iov.key = rma_op_data->remote_mr_key;
+
+	/* Initialize the message */
+	msg.msg_iov = &iov;
+	msg.desc = NULL;
+	msg.iov_count = 1;
+	msg.addr = comm_rail->remote_addr;
+	msg.rma_iov = &rma_iov;
+	msg.rma_iov_count = 1;
+	msg.context = req;
+	msg.data = 0;
```

Comment:
I'm curious what is the purpose of this `rma_write_data` op if you're setting the immediate data to 0. When you get the completion on the other side, how will you match this completion with the buffer to which data was written? Unless you're merely maintaining a count of the number of completed write events on the remote side?

If you don't need the completion on the remote side, wouldn't `post_rma_write_op` suffice?

Reply:
Ok, I misunderstood this function. `post_rma_write_data_op` is an inject write, not a write with completion data.

Can we rename this `post_rma_write_inject_op` and so on for the other references? Otherwise, it's easy to confuse with Libfabric's `fi_writedata` (write with immediate completion data), which is used elsewhere in the plugin.

Reply:
Renamed API function into `iwrite_inline`. Also renamed post functions like `post_rma_write_inject`, ...